### ISSUE TITLE
Document curriculum learning with worked examples

### ISSUE THREAD

**Issue by dreamlessx** [dreamlessx/LandmarkDiff-public#50]
## Problem

We have a `TrainingCurriculum` class and `CurriculumDataset` wrapper in `landmarkdiff/curriculum.py`, plus a config file `configs/phaseA_v3_curriculum.yaml`, but no documentation explaining how to actually use curriculum learning or why you would want to.

Someone looking at the config would see `warmup_fraction: 0.1` and `full_difficulty_at: 0.5` but have no idea what those numbers mean in practice, how to tune them, or what effect they have on convergence.

### 1. Conceptual guide (in docs/)

A page explaining:

- What curriculum learning does for LandmarkDiff specifically (start with small/easy displacements, gradually introduce large deformations)
- Why this helps: the ControlNet struggles early in training with large deformations because the conditioning signal is too far from the base image. Starting easy gives it time to learn the small-displacement manifold first.
- How the difficulty schedule works (cosine ramp between warmup and full_difficulty)
- How the metadata JSON maps displacement magnitude to difficulty scores

### 2. Worked example

A notebook or script showing:

```python
from landmarkdiff.curriculum import TrainingCurriculum

curriculum = TrainingCurriculum(
    total_steps=100000,
    warmup_fraction=0.1,
    full_difficulty_at=0.5,
)

# Show the difficulty curve
steps = range(0, 100000, 1000)
difficulties = [curriculum.get_difficulty(s) for s in steps]
# Plot it
```

Then show what the training data looks like at difficulty=0.2 vs difficulty=0.8 (small nose tweaks vs large orthognathic shifts).

### 3. Config reference

Document each field in the curriculum config section:

| Field | Type | Default | Description |
|-------|------|---------|-------------|
| `warmup_fraction` | float | 0.1 | Fraction of training spent at minimum difficulty |
| `full_difficulty_at` | float | 0.5 | Step fraction where full dataset is used |
| `difficulty_metric` | str | "displacement_magnitude" | How to rank sample difficulty |
| `min_difficulty_percentile` | float | 0.0 | Floor for filtering |

### 4. Tuning guidelines

From our experiments:
- `warmup_fraction=0.1` works well for 100k step runs
- Setting `full_difficulty_at` too early (< 0.3) leads to training instability around step 30k
- Setting it too late (> 0.7) wastes compute on easy samples
- For fine-tuning a pretrained model, you can skip warmup entirely

## Files involved

- `landmarkdiff/curriculum.py` - has good docstrings already, those can be pulled into the guide
- `configs/phaseA_v3_curriculum.yaml` - the reference config
- `scripts/launch_training.py` - where curriculum gets wired in

**Comment by dreamlessx** [comment]
Closing -- curriculum learning is documented in docs/tutorials/training.md and docs/GPU_TRAINING_GUIDE.md, with worked examples using the configs in configs/. The phaseA/B/C curriculum configs demonstrate step-by-step progression from simple to complex deformations with annotated YAML.
